### PR TITLE
feat: Specify config file path

### PR DESCRIPTION
**Pull request by kontax**
## Description

Add the ability to specify a configuration file outside of the default path using `-c, --config <CONFIG>`.

Forgive any crappy code, my rust is at a pretty noob stage.

## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

## How Has This Been Tested?

- [x] `cargo test`
- [x] Manually testing a configuration file outside the default location using the `-c` flag
- [x] Testing a configuration file gets generate automatically when specified and non-existant

I have not generated any automated tests for this.

### CODE REVIEW

File: src/config/mod.rs [hunk 1]
```diff
@@ -119,6 +116,20 @@ impl Config {
     }
 }
 
+pub fn get_config_path(args: &Option<String>) -> Result<PathBuf, Box<dyn Error>> {
+    match args {
```
Comment (cyrinux) [inline src/config/mod.rs:120]:
```suggestion
    let cfg_path = match args {
```

File: src/config/mod.rs [hunk 2]
```diff
@@ -119,6 +116,20 @@ impl Config {
...
+        Some(path) => {
+            let cfg_path = PathBuf::from(path);
+            Ok(cfg_path)
```
Comment (cyrinux) [inline src/config/mod.rs:123]:
```suggestion
```

File: src/config/mod.rs [hunk 3]
```diff
@@ -119,6 +116,20 @@ impl Config {
...
+        }
+        _ => {
+            let xdg_dirs = xdg::BaseDirectories::with_prefix("hyprland-autoname-workspaces")?;
+            let cfg_path = xdg_dirs.place_config_file("config.toml")?;
+            Ok(cfg_path)
```
Comment (cyrinux) [inline src/config/mod.rs:128]:
```suggestion
```

File: src/config/mod.rs [hunk 4]
```diff
@@ -119,6 +116,20 @@ impl Config {
...
 
+pub fn get_config_path(args: &Option<String>) -> Result<PathBuf, Box<dyn Error>> {
+    match args {
+        Some(path) => {
+            let cfg_path = PathBuf::from(path);
+            Ok(cfg_path)
+        }
+        _ => {
+            let xdg_dirs = xdg::BaseDirectories::with_prefix("hyprland-autoname-workspaces")?;
+            let cfg_path = xdg_dirs.place_config_file("config.toml")?;
```
Comment (cyrinux) [inline src/config/mod.rs:127]:
```suggestion
            xdg_dirs.place_config_file("config.toml")?
```

File: src/config/mod.rs [hunk 5]
```diff
@@ -119,6 +116,20 @@ impl Config {
     }
 }
 
+pub fn get_config_path(args: &Option<String>) -> Result<PathBuf, Box<dyn Error>> {
+    match args {
+        Some(path) => {
+            let cfg_path = PathBuf::from(path);
```
Comment (cyrinux) [inline src/config/mod.rs:122]:
```suggestion
            PathBuf::from(path)
```

File: src/config/mod.rs [hunk 6]
```diff
@@ -119,6 +116,20 @@ impl Config {
...
+            Ok(cfg_path)
+        }
+        _ => {
+            let xdg_dirs = xdg::BaseDirectories::with_prefix("hyprland-autoname-workspaces")?;
+            let cfg_path = xdg_dirs.place_config_file("config.toml")?;
+            Ok(cfg_path)
+        }
+    }
```
Comment (cyrinux) [inline src/config/mod.rs:130]:
```suggestion
    };
    Ok(cfg_path)
```

**Comment by cyrinux** [comment]
Thanks @kontax,

I neatpick about a preference, return just one time at the end, otherwise, looks good to me.
I will merge when I feel ready about. I got a bigger refactoring in progress 😉

**Comment by kontax** [comment]
Thanks @cyrinux, appreciate the feedback! Looking forward to the refactor, let me know if I can help at all!

**Comment by cyrinux** [comment]
I progress in the refactoring, and will manually, I will manually add your PR, we change too many thing 😉

**Comment by cyrinux** [comment]
Merge manually in https://github.com/hyprland-community/hyprland-autoname-workspaces/releases/tag/1.0.0

Thanks @kontax

**Pull request closed without merging**